src/format.rs

```diff
@@ -0,0 +1,888 @@
+        let n = self.width.unwrap_or(9).saturating_sub(9) + 3;
+
+        match self.padding {
+            Padding::Spaces => write!(
+                f,
+                "{: >+n$.0}:{:02}:{:02}",
+                utc_offset.hour, utc_offset.minute, utc_offset.second
+            ),
+            _ => write!(
+                f,
+                "{:+0n$.0}:{:02}:{:02}",
+                utc_offset.hour, utc_offset.minute, utc_offset.second
+            ),
+        }
+    }
+
+    fn write_padding(&self, f: &mut SizeLimiter<'_>, min_width: usize) -> io::Result<()> {
+        let n = self.width.unwrap_or(min_width).saturating_sub(min_width);
+
+        match self.padding {
+            Padding::Zeros => write!(f, "{:0>n$}", ""),
+            _ => write!(f, "{: >n$}", ""),
+        }
+    }
+
+    #[allow(clippy::too_many_lines)]
+    fn fmt(&self, f: &mut SizeLimiter<'_>, time: &Time) -> io::Result<()> {
+        match self.spec {
+            Spec::Year4Digits => {
+                let year = time.year();
+                let default_width = if year < 0 { 5 } else { 4 };
+                self.format_num_zeros(f, year, default_width)
+            }
+            Spec::YearDiv100 => self.format_num_zeros(f, time.year().div_euclid(100), 2),
+            Spec::YearRem100 => self.format_num_zeros(f, time.year().rem_euclid(100), 2),
+            Spec::Month => self.format_num_zeros(f, time.month(), 2),
+            Spec::MonthName => {
+                let index = (time.month() - 1) as usize;
+                match self.case {
+                    Some(_) => self.format_string(f, MONTHS_UPPER[index]),
+                    None => self.format_string(f, MONTHS[index]),
+                }
+            }
+            Spec::MonthNameAbbr => {
+                let index = (time.month() - 1) as usize;
+                match self.case {
+                    Some(_) => self.format_string(f, &MONTHS_UPPER[index][..3]),
+                    None => self.format_string(f, &MONTHS[index][..3]),
+                }
+            }
+            Spec::MonthDayZero => self.format_num_zeros(f, time.day(), 2),
+            Spec::MonthDaySpace => self.format_num_spaces(f, time.day(), 2),
+            Spec::YearDay => self.format_num_zeros(f, time.day_of_year(), 3),
+            Spec::Hour24hZero => self.format_num_zeros(f, time.hour(), 2),
+            Spec::Hour24hSpace => self.format_num_spaces(f, time.hour(), 2),
+            Spec::Hour12hZero => {
+                let mut hour = time.hour() % 12;
+                hour = if hour == 0 { 12 } else { hour };
+                self.format_num_zeros(f, hour, 2)
+            }
+            Spec::Hour12hSpace => {
+                let mut hour = time.hour() % 12;
+                hour = if hour == 0 { 12 } else { hour };
+                self.format_num_spaces(f, hour, 2)
+            }
+            Spec::MeridianLower => {
+                let (am, pm) = match self.case {
+                    Some(_) => ("AM", "PM"),
```

Comment:
I think I'd prefer to have the enum variants listed in the `Some` to make the code a bit more easy to reason about locally. I had to scan back to the `Case` enum to figure out what was happening.

```suggestion
                    Some(Case::Change | Case::Upper) => ("AM", "PM"),
```